File: python/examples/quickstart_scrape.py
"""Quickstart: scrape a single page.

Run with:
    GHOSTCRAWL_API_KEY=gck_live_YOUR_KEY python examples/quickstart_scrape.py
"""

from __future__ import annotations

import asyncio
import os
import sys


async def main() -> None:
    token = os.environ.get("GHOSTCRAWL_API_KEY")
    if not token:
        print("Set GHOSTCRAWL_API_KEY in your environment.", file=sys.stderr)
        sys.exit(1)

    from ghostcrawl import GhostCrawlClient

    async with GhostCrawlClient(token=token) as client:
        result = await client.scrape(url="https://example.com", format="markdown")
        print("Scrape status:", result.get("status", "unknown"))
        # identity_id (Phase 140.4-16 response envelope field) — printed so a caller can
        # correlate this exact drive to its server-side egress-exit assignment (D-04, phase 177).
        print("identity_id:", result.get("identity_id", "unknown"))
        # The API returns content in the field matching the requested format.
        # format="markdown" → result["markdown"]; format="html" → result["html"]
        content = result.get("markdown") or result.get("content") or result.get("html", "")
        print("Content preview:")
        print(content[:500] if isinstance(content, str) else str(content)[:500])


if __name__ == "__main__":
    asyncio.run(main())

File: node/examples/quickstart-scrape.ts
/**
 * Quickstart: scrape a single URL and print the content.
 *
 * Run with:
 *   GHOSTCRAWL_API_KEY=gck_live_YOUR_KEY npx tsx examples/quickstart-scrape.ts
 */
import { createGhostCrawlClient } from "../src/index.js";

async function main(): Promise<void> {
  const token = process.env.GHOSTCRAWL_API_KEY;
  if (!token) {
    console.error("Set GHOSTCRAWL_API_KEY in your environment.");
    process.exit(1);
  }

  const client = createGhostCrawlClient({ token });

  // Canonical LLM-ready markdown scrape. On the cloud a no-identity scrape is served by
  // an auto-assigned SPOOFED identity, and the markdown envelope carries identity_id so
  // this exact drive can be correlated to its egress-exit assignment (D-04).
  const result = await client.scrape({
    url: "https://example.com",
    format: "markdown",
  });

  console.log("Scrape status:", result["status"] ?? "unknown");
  console.log("identity_id:", result["identity_id"] ?? "unknown");
  const content = result["markdown"] ?? result["content"] ?? result["html"] ?? "";
  console.log("Content preview:");
  console.log(String(content).slice(0, 500));
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
